scripts/romanisim-make-stack

```diff
@@ -0,0 +1,267 @@
+
+    # Create script making argument
+    parser.add_argument('-m', "--make_script",
+                        action='store_true',
+                        help='Create list of romanisim calls from csv file (e.g. for cluster usage).')
+
+    # Number of objects to simulate
+    parser.add_argument('-n', '--nobj',
+                        type=int,
+                        default=1000,
+                        help='Specify number of objects to simulate.')
+
+    # Persistence
+    parser.add_argument('-p', '--previous',
+                        type=str,
+                        metavar='previous.asdf',
+                        nargs=1,
+                        help=(
+                            'Previous simulated file in chronological order used for persistence modeling.'))
+
+    # Random number seed
+    parser.add_argument('-r', '--rng_seed',
+                        type=int,
+                        default=None,
+                        help='Random number seed (int)')
+
+    # SCA detector identifier number
+    parser.add_argument('-s','--sca',
+                        type=int,
+                        default=0,
+                        help='SCA to simulate')
+
+    # Use CRDS for distortion maps
+    parser.add_argument('-u', '--usecrds',
+                        action='store_true',
+                        help='Use CRDS for distortion map')
+
+    # Use CRDS stored PSFs
+    parser.add_argument('-w', '--webbpsf',
+                        action='store_true',
+                        help='Use webbpsf for PSFs.')
+
+
+
+
+    args = parser.parse_args()
+
+    log.info('Starting simulation...')
+    log.warning("romanisim is under active development.  Its output has "
+                "not been formally validated; only limited testing has been "
+                "performed.  For this reason, use of romanisim for "
+                "preparation of ROSES proposals is not advised.  Other "
+                "packages like galsim's roman package or STIPS may better "
+                "serve such purposes.")
+
+    # Set up metadata
+    metadata = deepcopy(default_parameters_dictionary)
+
+    # If creating a script, create the script file
+    if args.make_script:
+        spcript_file = open("romansim_stack.script", "w")
+
+        # Acquire cleaned options to pass to script call
+        options_dct = {}
+        for option, value in vars(args).items():
+            if args.debug:
+                print(f'option, value = {option, value}')
+            if option not in ('input_file_name','debug','make_script') and value is not None:
+                options_dct[option] = value
+
+
+
+    with open(args.input_file_name, newline='') as infile:
+        reader = csv.DictReader(infile, delimiter=',', skipinitialspace=True)
+        for entry in reader:
+            # Debug: print keys
+            if args.debug:
+                print(f'entry.keys = {entry.keys()}')
+
+            # Reset time offset
+            time_offset = 0 * u.s
```

Comment:
This gets called every time and isn't what we want?

Reply:
It should be, as the offset is reset for each new entry (and then added to per exposure).

Reply:
entry_idx should be exp_idx (exposure index); the time offset grows with each exposure and then gets reset once we go to the next entry / line.

Reply:
"In the middle loop supposed to be, e.g., taking 10 exposures at the same place?"

This is exactly it. I figured that was what we were going to be given, otherwise, we wouldn't need overhead time (since if they listed one per line for every exposure, the time would match the start time of each). FYI, here is the input CSV I have been using:

```
ra, dec, roll_angle, optical_element, date, exposures, overhead_time, ma_table_number
270.0, 66.0, 0, F087, 2026-01-01T00:00:00, 4, 90, 1
270.0, 66.0, 45, F213, 2026-01-02T00:00:00, 2, 300, 1
273.0, 62.0, 10, F184, 2026-02-01T00:00:00, 3, 90, 1
273.0, 62.0, 55, F158, 2026-02-04T00:00:00, 1, 120, 1
270.0, 66.0, 0, F146, 2026-06-01T00:00:00, 2, 90, 1
270.0, 66.0, 60, F062, 2026-06-01T12:00:00, 4, 180, 1
273.0, 66.0, 10, F129, 2026-06-02T00:00:00, 3, 90, 1
273.0, 66.0, 55, F106, 2026-06-02T08:00:00, 1, 120, 1
```

Reply:
Thanks, okay.

Let's aim for this format:
https://jira.stsci.edu/browse/APT-93485
In the context of that format, the 'duration' is the number of seconds that the exposure takes plus any overheads. I want to avoid our thinking about things like time needed to change the filter wheel, or how fast Roman can slew or spin or whatever, or how long settling + guide star acquisition + ... takes.

So we'll take one --date on the command line and add the duration to it each time to get the date of the next exposure.

You're right that an alternative would be to just do a separate date on each row.  Let's follow what I told them, though.  Let's not do repeated exposures at each location in either case.

Reply:
Do you want date in the integer format of romanisim-make-image, UTC string, or something else? I am obviously a fan of the latter, but I will defer to your preference.